stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/builtin/systemlib/Sprintf.java

```diff
@@ -0,0 +1,113 @@
+        // k records number of format specifiers seen so far, used to read respective array element
+        for (int i = 0, j = 0, k = 0; i < format.length(); i++) {
+            if (format.charAt(i) == '%' && ((i+1) < format.length())) {
+
+                // skip % character
+                j = i + 1;
+
+                if (k >= args.size()) {
+                    // there's not enough arguments
+                    throw BLangExceptionHelper.getRuntimeException(
+                            RuntimeErrors.NOT_ENOUGH_FORMAT_ARGUMENTS);
+                }
+                StringBuilder padding = new StringBuilder();
+                for(; Character.isDigit(format.charAt(j)) || format.charAt(j) == '.'; j++) {
+                    padding.append(format.charAt(j));
+                }
+                switch (format.charAt(j)) {
+                    case 'b': result.append(String.format("%" + padding + "b", args.get(k).value()));          break;
+                    case 'd': result.append(String.format("%" + padding + "d", args.get(k).value()));          break;
+                    case 'f': result.append(String.format("%" + padding + "f", args.get(k).value()));          break;
+                    case 's': result.append(String.format("%" + padding + "s", args.get(k).value()));          break;
+                    case 'x': result.append(String.format("%" + padding + "x", args.get(k).value()));          break;
+                    case 'X': result.append(String.format("%" + padding + "X", args.get(k).value()));          break;
+                    case 'o': result.append(String.format("%" + padding + "o", args.get(k).value()));          break;
+                    case 'B': result.append(Integer.toBinaryString(Integer.parseInt(args.get(k).stringValue())));                      break;
+                    case 'm': // fall through
+                    case 'e':
+                    case 'p':
+                    case 'l':
+                    case 'L':
+                    case 'j':
+                    case 't':
+                    case 'r':
+                    case 'a': result.append(String.format("%" + padding + "s", args.get(k).stringValue()));    break;
+                    case '%': result.append("%"); break;
+                    default:
+                        // format string not supported
+                        throw BLangExceptionHelper.getRuntimeException(
+                                RuntimeErrors.INVALID_FORMAT_SPECIFIER, format.charAt(j));
+                }
+                if (format.charAt(j) == '%') {
+                    // special case %%, don't count as a format specifier
+                    i++;
+                } else {
+                    k++;
+                    i = j;
+                }
+                continue;
+            }
+            // no match, copy and continue
+            result.append(format.charAt(i));
+        }
+        return getBValues(new BString(result.toString()));
+    }
+}
```

Comment:
Shall we reformat the code. Having issues which causes check-style failures.

Reply:
No formatting necessary for this line. No issues with checkstyle